File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/questionanswer/CommittedAnswerImplV1_0.java
package com.evernym.verity.sdk.protocols.questionanswer;

import com.evernym.verity.sdk.protocols.questionanswer.v1_0.CommittedAnswerV1_0;

/*
 * NON_VISIBLE
 *
 * This is an implementation of CommittedAnswerImplV1_0 but is not viable to user of Verity SDK. Created using the
 * static CommittedAnswer class
 */
class CommittedAnswerImplV1_0 extends AskCommonImpl implements CommittedAnswerV1_0 {

    CommittedAnswerImplV1_0(String forRelationship,
                                   String questionText,
                                   String questionDetail,
                                   String[] validResponses,
                                   Boolean signatureRequired) {
        super(forRelationship, questionText, questionDetail, validResponses, signatureRequired);
    }

    CommittedAnswerImplV1_0(String forRelationship, String threadId, String answer) {
        super(forRelationship, threadId, answer);
    }

    CommittedAnswerImplV1_0(String forRelationship, String threadId) {
        super(forRelationship, threadId);
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/QuestionAnswer/CommittedAnswerImplV1_0.cs
using System;
using System.Json;

namespace VeritySDK
{
    /// <summary>
    /// This is an implementation of CommittedAnswerImplV1_0 but is not viable to user of Verity SDK. Created using the static CommittedAnswer class 
    /// </summary>
    public class CommittedAnswerImplV1_0 : CommittedAnswerV1_0
    {
        public CommittedAnswerImplV1_0(string forRelationship,
                                   string questionText,
                                   string questionDetail,
                                   string[] validResponses,
                                   bool signatureRequired) : base(forRelationship, questionText, questionDetail, validResponses, signatureRequired) { }

        public CommittedAnswerImplV1_0(string forRelationship, string threadId, string answer) : base(forRelationship, threadId, answer) { }

        public CommittedAnswerImplV1_0(string forRelationship, string threadId) : base(forRelationship, threadId) { }
    }
}